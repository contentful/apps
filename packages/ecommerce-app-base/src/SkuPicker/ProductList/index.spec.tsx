import React from 'react';
import { render, cleanup, configure } from '@testing-library/react';
import { Props, ProductList } from '.';
import productPreviews from '../../__mocks__/productPreviews';

configure({
  testIdAttribute: 'data-test-id',
});

const defaultProps: Props = {
  products: productPreviews,
  selectProduct: jest.fn(),
  selectedSKUs: [],
};

const renderComponent = (props: Props) => {
  return render(<ProductList {...props} />);
};

describe('ProductList', () => {
  afterEach(cleanup);

  it('should render successfully with no items selected', async () => {
    const { getAllByTestId, getByTestId } = renderComponent(defaultProps);
    expect(getAllByTestId('ProductListItem')).toHaveLength(productPreviews.length);
    for (let product of productPreviews) {
      expect(getByTestId(`product-preview-${product.sku}`)).toHaveAttribute(
        'aria-checked',
        'false'
      );
    }
  });

  it('should render successfully with selected items', () => {
    const selectedSKU = productPreviews[1].sku;
    const { getAllByTestId, getByTestId } = renderComponent({
      ...defaultProps,
      selectedSKUs: [selectedSKU],
    });
    expect(getAllByTestId('ProductListItem')).toHaveLength(productPreviews.length);
    for (let product of productPreviews) {
      expect(getByTestId(`product-preview-${product.sku}`)).toHaveAttribute(
        'aria-checked',
        selectedSKU === product.sku ? 'true' : 'false'
      );
    }
  });
});
