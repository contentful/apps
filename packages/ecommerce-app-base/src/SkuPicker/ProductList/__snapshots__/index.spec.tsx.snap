// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList should render successfully with no items selected 1`] = `
<div>
  <div
    class="css-almiqb"
  >
    <div
      class="css-jfu8ng"
    >
      <div
        aria-checked="false"
        class="css-16t4lg0 "
        data-test-id="product-preview-M0E20130820E90Z"
        role="switch"
        tabindex="-1"
      >
        <svg
          aria-label="Loading component..."
          class="SkeletonContainer__SkeletonContainer___23jiu css-7uqnex"
          data-test-id="cf-ui-skeleton-form"
          height="100%"
          preserveAspectRatio="xMidYMid meet"
          role="img"
          width="100%"
        >
          <title>
            Loading component...
          </title>
          <rect
            clip-path="url(#cf-ui-skeleton-clip-0)"
            height="100%"
            style="fill: url(#cf-ui-skeleton-clip-gradient-1);"
            width="100%"
            x="0"
            y="0"
          />
          <defs>
            <clippath
              id="cf-ui-skeleton-clip-0"
            >
              <rect
                data-test-id="cf-ui-skeleton-image"
                height="290"
                rx="0"
                ry="0"
                width="400"
              />
            </clippath>
            <lineargradient
              id="cf-ui-skeleton-clip-gradient-1"
            >
              <stop
                offset="0%"
                stop-color="#e5ebed"
                stop-opacity="1"
              >
                <animate
                  attributeName="stop-color"
                  dur="2s"
                  id="animation-2"
                  repeatCount="indefinite"
                  values="#e5ebed; #f7f9fa; #e5ebed"
                />
              </stop>
              <stop
                offset="50%"
                stop-color="#f7f9fa"
                stop-opacity="1"
              >
                <animate
                  attributeName="stop-color"
                  begin="animation-2.begin+0.25s"
                  dur="2s"
                  repeatCount="indefinite"
                  values="#e5ebed; #f7f9fa; #e5ebed"
                />
              </stop>
              <stop
                offset="100%"
                stop-color="#e5ebed"
                stop-opacity="1"
              >
                <animate
                  attributeName="stop-color"
                  begin="animation-2.begin+0.5s"
                  dur="2s"
                  repeatCount="indefinite"
                  values="#e5ebed; #f7f9fa; #e5ebed"
                />
              </stop>
            </lineargradient>
          </defs>
        </svg>
        <div
          class="css-nk39el"
        >
          <img
            alt="product preview"
            class="css-1ddetr"
            data-test-id="image"
            src="https://example.com/commercetools-example/products/picture-1.jpg"
            style="display: none;"
          />
          <img
            alt="check"
            class="css-1imdisq"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDQiIGhlaWdodD0iNDQiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+PGcgZmlsbD0iIzMwNzJCRSIgZmlsbC1ydWxlPSJldmVub2RkIj48Y2lyY2xlIGZpbGwtb3BhY2l0eT0iLjEiIGN4PSIyMiIgY3k9IjIyIiByPSIyMiIvPjxwYXRoIGQ9Ik0zMS42ODkgMTYuNjE0TDI5LjU1NSAxNC41IDE5LjAyNCAyNS4yNzVsLTQuNjItNC40MDYtMi4wOTIgMi4xNTMgNi43NSA2LjQ3OCAxMi42MjctMTIuODg2eiIgZmlsbC1ydWxlPSJub256ZXJvIi8+PC9nPjwvc3ZnPg=="
          />
        </div>
        <p
          class="css-182natc"
        >
          Dress Twin-Set rose
        </p>
        <p
          class="css-1bgao96"
        >
          M0E20130820E90Z
        </p>
      </div>
    </div>
    <div
      class="css-jfu8ng"
    >
      <div
        aria-checked="false"
        class="css-16t4lg0 "
        data-test-id="product-preview-A0E2300FX102203"
        role="switch"
        tabindex="-1"
      >
        <svg
          aria-label="Loading component..."
          class="SkeletonContainer__SkeletonContainer___23jiu css-7uqnex"
          data-test-id="cf-ui-skeleton-form"
          height="100%"
          preserveAspectRatio="xMidYMid meet"
          role="img"
          width="100%"
        >
          <title>
            Loading component...
          </title>
          <rect
            clip-path="url(#cf-ui-skeleton-clip-3)"
            height="100%"
            style="fill: url(#cf-ui-skeleton-clip-gradient-4);"
            width="100%"
            x="0"
            y="0"
          />
          <defs>
            <clippath
              id="cf-ui-skeleton-clip-3"
            >
              <rect
                data-test-id="cf-ui-skeleton-image"
                height="290"
                rx="0"
                ry="0"
                width="400"
              />
            </clippath>
            <lineargradient
              id="cf-ui-skeleton-clip-gradient-4"
            >
              <stop
                offset="0%"
                stop-color="#e5ebed"
                stop-opacity="1"
              >
                <animate
                  attributeName="stop-color"
                  dur="2s"
                  id="animation-5"
                  repeatCount="indefinite"
                  values="#e5ebed; #f7f9fa; #e5ebed"
                />
              </stop>
              <stop
                offset="50%"
                stop-color="#f7f9fa"
                stop-opacity="1"
              >
                <animate
                  attributeName="stop-color"
                  begin="animation-5.begin+0.25s"
                  dur="2s"
                  repeatCount="indefinite"
                  values="#e5ebed; #f7f9fa; #e5ebed"
                />
              </stop>
              <stop
                offset="100%"
                stop-color="#e5ebed"
                stop-opacity="1"
              >
                <animate
                  attributeName="stop-color"
                  begin="animation-5.begin+0.5s"
                  dur="2s"
                  repeatCount="indefinite"
                  values="#e5ebed; #f7f9fa; #e5ebed"
                />
              </stop>
            </lineargradient>
          </defs>
        </svg>
        <div
          class="css-nk39el"
        >
          <img
            alt="product preview"
            class="css-1ddetr"
            data-test-id="image"
            src="https://example.com/commercetools-example/products/picture-2.jpg"
            style="display: none;"
          />
          <img
            alt="check"
            class="css-1imdisq"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDQiIGhlaWdodD0iNDQiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+PGcgZmlsbD0iIzMwNzJCRSIgZmlsbC1ydWxlPSJldmVub2RkIj48Y2lyY2xlIGZpbGwtb3BhY2l0eT0iLjEiIGN4PSIyMiIgY3k9IjIyIiByPSIyMiIvPjxwYXRoIGQ9Ik0zMS42ODkgMTYuNjE0TDI5LjU1NSAxNC41IDE5LjAyNCAyNS4yNzVsLTQuNjItNC40MDYtMi4wOTIgMi4xNTMgNi43NSA2LjQ3OCAxMi42MjctMTIuODg2eiIgZmlsbC1ydWxlPSJub256ZXJvIi8+PC9nPjwvc3ZnPg=="
          />
        </div>
        <p
          class="css-182natc"
        >
          Michael Kors – Shopper “Jet Set Travel”
        </p>
        <p
          class="css-1bgao96"
        >
          A0E2300FX102203
        </p>
      </div>
    </div>
    <div
      class="css-jfu8ng"
    >
      <div
        aria-checked="false"
        class="css-16t4lg0 "
        data-test-id="product-preview-M0E21300900DZN7"
        role="switch"
        tabindex="-1"
      >
        <svg
          aria-label="Loading component..."
          class="SkeletonContainer__SkeletonContainer___23jiu css-7uqnex"
          data-test-id="cf-ui-skeleton-form"
          height="100%"
          preserveAspectRatio="xMidYMid meet"
          role="img"
          width="100%"
        >
          <title>
            Loading component...
          </title>
          <rect
            clip-path="url(#cf-ui-skeleton-clip-6)"
            height="100%"
            style="fill: url(#cf-ui-skeleton-clip-gradient-7);"
            width="100%"
            x="0"
            y="0"
          />
          <defs>
            <clippath
              id="cf-ui-skeleton-clip-6"
            >
              <rect
                data-test-id="cf-ui-skeleton-image"
                height="290"
                rx="0"
                ry="0"
                width="400"
              />
            </clippath>
            <lineargradient
              id="cf-ui-skeleton-clip-gradient-7"
            >
              <stop
                offset="0%"
                stop-color="#e5ebed"
                stop-opacity="1"
              >
                <animate
                  attributeName="stop-color"
                  dur="2s"
                  id="animation-8"
                  repeatCount="indefinite"
                  values="#e5ebed; #f7f9fa; #e5ebed"
                />
              </stop>
              <stop
                offset="50%"
                stop-color="#f7f9fa"
                stop-opacity="1"
              >
                <animate
                  attributeName="stop-color"
                  begin="animation-8.begin+0.25s"
                  dur="2s"
                  repeatCount="indefinite"
                  values="#e5ebed; #f7f9fa; #e5ebed"
                />
              </stop>
              <stop
                offset="100%"
                stop-color="#e5ebed"
                stop-opacity="1"
              >
                <animate
                  attributeName="stop-color"
                  begin="animation-8.begin+0.5s"
                  dur="2s"
                  repeatCount="indefinite"
                  values="#e5ebed; #f7f9fa; #e5ebed"
                />
              </stop>
            </lineargradient>
          </defs>
        </svg>
        <div
          class="css-nk39el"
        >
          <img
            alt="product preview"
            class="css-1ddetr"
            data-test-id="image"
            src="https://example.com/commercetools-example/products/picture-3.jpg"
            style="display: none;"
          />
          <img
            alt="check"
            class="css-1imdisq"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDQiIGhlaWdodD0iNDQiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+PGcgZmlsbD0iIzMwNzJCRSIgZmlsbC1ydWxlPSJldmVub2RkIj48Y2lyY2xlIGZpbGwtb3BhY2l0eT0iLjEiIGN4PSIyMiIgY3k9IjIyIiByPSIyMiIvPjxwYXRoIGQ9Ik0zMS42ODkgMTYuNjE0TDI5LjU1NSAxNC41IDE5LjAyNCAyNS4yNzVsLTQuNjItNC40MDYtMi4wOTIgMi4xNTMgNi43NSA2LjQ3OCAxMi42MjctMTIuODg2eiIgZmlsbC1ydWxlPSJub256ZXJvIi8+PC9nPjwvc3ZnPg=="
          />
        </div>
        <p
          class="css-182natc"
        >
          Shirt “Jenny“ Polo Ralph Lauren white
        </p>
        <p
          class="css-1bgao96"
        >
          M0E21300900DZN7
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`ProductList should render successfully with selected items 1`] = `
<div>
  <div
    class="css-almiqb"
  >
    <div
      class="css-jfu8ng"
    >
      <div
        aria-checked="false"
        class="css-16t4lg0 "
        data-test-id="product-preview-M0E20130820E90Z"
        role="switch"
        tabindex="-1"
      >
        <svg
          aria-label="Loading component..."
          class="SkeletonContainer__SkeletonContainer___23jiu css-7uqnex"
          data-test-id="cf-ui-skeleton-form"
          height="100%"
          preserveAspectRatio="xMidYMid meet"
          role="img"
          width="100%"
        >
          <title>
            Loading component...
          </title>
          <rect
            clip-path="url(#cf-ui-skeleton-clip-9)"
            height="100%"
            style="fill: url(#cf-ui-skeleton-clip-gradient-10);"
            width="100%"
            x="0"
            y="0"
          />
          <defs>
            <clippath
              id="cf-ui-skeleton-clip-9"
            >
              <rect
                data-test-id="cf-ui-skeleton-image"
                height="290"
                rx="0"
                ry="0"
                width="400"
              />
            </clippath>
            <lineargradient
              id="cf-ui-skeleton-clip-gradient-10"
            >
              <stop
                offset="0%"
                stop-color="#e5ebed"
                stop-opacity="1"
              >
                <animate
                  attributeName="stop-color"
                  dur="2s"
                  id="animation-11"
                  repeatCount="indefinite"
                  values="#e5ebed; #f7f9fa; #e5ebed"
                />
              </stop>
              <stop
                offset="50%"
                stop-color="#f7f9fa"
                stop-opacity="1"
              >
                <animate
                  attributeName="stop-color"
                  begin="animation-11.begin+0.25s"
                  dur="2s"
                  repeatCount="indefinite"
                  values="#e5ebed; #f7f9fa; #e5ebed"
                />
              </stop>
              <stop
                offset="100%"
                stop-color="#e5ebed"
                stop-opacity="1"
              >
                <animate
                  attributeName="stop-color"
                  begin="animation-11.begin+0.5s"
                  dur="2s"
                  repeatCount="indefinite"
                  values="#e5ebed; #f7f9fa; #e5ebed"
                />
              </stop>
            </lineargradient>
          </defs>
        </svg>
        <div
          class="css-nk39el"
        >
          <img
            alt="product preview"
            class="css-1ddetr"
            data-test-id="image"
            src="https://example.com/commercetools-example/products/picture-1.jpg"
            style="display: none;"
          />
          <img
            alt="check"
            class="css-1imdisq"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDQiIGhlaWdodD0iNDQiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+PGcgZmlsbD0iIzMwNzJCRSIgZmlsbC1ydWxlPSJldmVub2RkIj48Y2lyY2xlIGZpbGwtb3BhY2l0eT0iLjEiIGN4PSIyMiIgY3k9IjIyIiByPSIyMiIvPjxwYXRoIGQ9Ik0zMS42ODkgMTYuNjE0TDI5LjU1NSAxNC41IDE5LjAyNCAyNS4yNzVsLTQuNjItNC40MDYtMi4wOTIgMi4xNTMgNi43NSA2LjQ3OCAxMi42MjctMTIuODg2eiIgZmlsbC1ydWxlPSJub256ZXJvIi8+PC9nPjwvc3ZnPg=="
          />
        </div>
        <p
          class="css-182natc"
        >
          Dress Twin-Set rose
        </p>
        <p
          class="css-1bgao96"
        >
          M0E20130820E90Z
        </p>
      </div>
    </div>
    <div
      class="css-jfu8ng"
    >
      <div
        aria-checked="true"
        class="css-16t4lg0 css-1rtg1va"
        data-test-id="product-preview-A0E2300FX102203"
        role="switch"
        tabindex="-1"
      >
        <svg
          aria-label="Loading component..."
          class="SkeletonContainer__SkeletonContainer___23jiu css-7uqnex"
          data-test-id="cf-ui-skeleton-form"
          height="100%"
          preserveAspectRatio="xMidYMid meet"
          role="img"
          width="100%"
        >
          <title>
            Loading component...
          </title>
          <rect
            clip-path="url(#cf-ui-skeleton-clip-12)"
            height="100%"
            style="fill: url(#cf-ui-skeleton-clip-gradient-13);"
            width="100%"
            x="0"
            y="0"
          />
          <defs>
            <clippath
              id="cf-ui-skeleton-clip-12"
            >
              <rect
                data-test-id="cf-ui-skeleton-image"
                height="290"
                rx="0"
                ry="0"
                width="400"
              />
            </clippath>
            <lineargradient
              id="cf-ui-skeleton-clip-gradient-13"
            >
              <stop
                offset="0%"
                stop-color="#e5ebed"
                stop-opacity="1"
              >
                <animate
                  attributeName="stop-color"
                  dur="2s"
                  id="animation-14"
                  repeatCount="indefinite"
                  values="#e5ebed; #f7f9fa; #e5ebed"
                />
              </stop>
              <stop
                offset="50%"
                stop-color="#f7f9fa"
                stop-opacity="1"
              >
                <animate
                  attributeName="stop-color"
                  begin="animation-14.begin+0.25s"
                  dur="2s"
                  repeatCount="indefinite"
                  values="#e5ebed; #f7f9fa; #e5ebed"
                />
              </stop>
              <stop
                offset="100%"
                stop-color="#e5ebed"
                stop-opacity="1"
              >
                <animate
                  attributeName="stop-color"
                  begin="animation-14.begin+0.5s"
                  dur="2s"
                  repeatCount="indefinite"
                  values="#e5ebed; #f7f9fa; #e5ebed"
                />
              </stop>
            </lineargradient>
          </defs>
        </svg>
        <div
          class="css-nk39el"
        >
          <img
            alt="product preview"
            class="css-1ddetr"
            data-test-id="image"
            src="https://example.com/commercetools-example/products/picture-2.jpg"
            style="display: none;"
          />
          <img
            alt="check"
            class="css-1bpg4ef"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDQiIGhlaWdodD0iNDQiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+PGcgZmlsbD0iIzMwNzJCRSIgZmlsbC1ydWxlPSJldmVub2RkIj48Y2lyY2xlIGZpbGwtb3BhY2l0eT0iLjEiIGN4PSIyMiIgY3k9IjIyIiByPSIyMiIvPjxwYXRoIGQ9Ik0zMS42ODkgMTYuNjE0TDI5LjU1NSAxNC41IDE5LjAyNCAyNS4yNzVsLTQuNjItNC40MDYtMi4wOTIgMi4xNTMgNi43NSA2LjQ3OCAxMi42MjctMTIuODg2eiIgZmlsbC1ydWxlPSJub256ZXJvIi8+PC9nPjwvc3ZnPg=="
          />
        </div>
        <p
          class="css-182natc"
        >
          Michael Kors – Shopper “Jet Set Travel”
        </p>
        <p
          class="css-1bgao96"
        >
          A0E2300FX102203
        </p>
      </div>
    </div>
    <div
      class="css-jfu8ng"
    >
      <div
        aria-checked="false"
        class="css-16t4lg0 "
        data-test-id="product-preview-M0E21300900DZN7"
        role="switch"
        tabindex="-1"
      >
        <svg
          aria-label="Loading component..."
          class="SkeletonContainer__SkeletonContainer___23jiu css-7uqnex"
          data-test-id="cf-ui-skeleton-form"
          height="100%"
          preserveAspectRatio="xMidYMid meet"
          role="img"
          width="100%"
        >
          <title>
            Loading component...
          </title>
          <rect
            clip-path="url(#cf-ui-skeleton-clip-15)"
            height="100%"
            style="fill: url(#cf-ui-skeleton-clip-gradient-16);"
            width="100%"
            x="0"
            y="0"
          />
          <defs>
            <clippath
              id="cf-ui-skeleton-clip-15"
            >
              <rect
                data-test-id="cf-ui-skeleton-image"
                height="290"
                rx="0"
                ry="0"
                width="400"
              />
            </clippath>
            <lineargradient
              id="cf-ui-skeleton-clip-gradient-16"
            >
              <stop
                offset="0%"
                stop-color="#e5ebed"
                stop-opacity="1"
              >
                <animate
                  attributeName="stop-color"
                  dur="2s"
                  id="animation-17"
                  repeatCount="indefinite"
                  values="#e5ebed; #f7f9fa; #e5ebed"
                />
              </stop>
              <stop
                offset="50%"
                stop-color="#f7f9fa"
                stop-opacity="1"
              >
                <animate
                  attributeName="stop-color"
                  begin="animation-17.begin+0.25s"
                  dur="2s"
                  repeatCount="indefinite"
                  values="#e5ebed; #f7f9fa; #e5ebed"
                />
              </stop>
              <stop
                offset="100%"
                stop-color="#e5ebed"
                stop-opacity="1"
              >
                <animate
                  attributeName="stop-color"
                  begin="animation-17.begin+0.5s"
                  dur="2s"
                  repeatCount="indefinite"
                  values="#e5ebed; #f7f9fa; #e5ebed"
                />
              </stop>
            </lineargradient>
          </defs>
        </svg>
        <div
          class="css-nk39el"
        >
          <img
            alt="product preview"
            class="css-1ddetr"
            data-test-id="image"
            src="https://example.com/commercetools-example/products/picture-3.jpg"
            style="display: none;"
          />
          <img
            alt="check"
            class="css-1imdisq"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDQiIGhlaWdodD0iNDQiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+PGcgZmlsbD0iIzMwNzJCRSIgZmlsbC1ydWxlPSJldmVub2RkIj48Y2lyY2xlIGZpbGwtb3BhY2l0eT0iLjEiIGN4PSIyMiIgY3k9IjIyIiByPSIyMiIvPjxwYXRoIGQ9Ik0zMS42ODkgMTYuNjE0TDI5LjU1NSAxNC41IDE5LjAyNCAyNS4yNzVsLTQuNjItNC40MDYtMi4wOTIgMi4xNTMgNi43NSA2LjQ3OCAxMi42MjctMTIuODg2eiIgZmlsbC1ydWxlPSJub256ZXJvIi8+PC9nPjwvc3ZnPg=="
          />
        </div>
        <p
          class="css-182natc"
        >
          Shirt “Jenny“ Polo Ralph Lauren white
        </p>
        <p
          class="css-1bgao96"
        >
          M0E21300900DZN7
        </p>
      </div>
    </div>
  </div>
</div>
`;
